respect replace_state on external navigation where applicable (#13303)
sveltejs/kit: .changeset/dirty-rats-divide.md

---
'@sveltejs/kit': patch
---

> respect `replaceState` on external navigation

Reviewer: ```suggestion
fix: respect `replaceState` on external navigation
```

> fix: respect `replaceState` on external navigation

Reviewer 2: ```suggestion
fix: respect `replaceState` when using `goto` results in an external navigation
```

---
'@sveltejs/kit': patch
---

> fix: respect `replaceState` when using `goto` results in an external navigation
> fix: respect `replaceState` when using `goto` in redirection chain

Reviewer 2: ```suggestion
fix: respect `replaceState` when using `goto` results in a redirect or an external navigation
```
